### PR TITLE
Add option (disabled by default) to disable idle watcher on fullscreen.

### LINKED ISSUE #224: idle watcher should be disabled on fullscreen
If you are watching a movie or doing something fullscreen, your idle time will increase very fast. I think it should be disabled.

##### Expected Behavior

idle watcher stops on fullscreen

##### Current Behavior

idle watcher works normally also on fullscreen

##### Possible Solution

I'm not sure about the best solution here. Add an option to set this behavior (enabled by default) or just implement it.
I can provide a patch to implement it when the right approach has been decided.

### PR DESCRIPTION
**Pull request by adjamhub**
Fixes #224
I also removed deprecated KIdleTime::timeoutReached(int) method and
used the actual timeoutReached(int,int) instead.

### CODE REVIEW

File: src/idlenesswatcher.cpp [hunk 1]
```diff
@@ -102,8 +104,18 @@ void IdlenessWatcher::setup()
...
-void IdlenessWatcher::timeoutReached(int identifier)
+void IdlenessWatcher::timeoutReached(int identifier,int /*msec*/)
 {
+    // check if disable Idleness when fullscreen is enabled
+    if (mPSettings.isDisableIdlenessWhenFullscreenEnabled()) {
+        WId w = KWindowSystem::activeWindow();
+        KWindowInfo info(w, NET::WMState);
+        if (info.hasState(NET::FullScreen)) {
+            qDebug() << "idleness timeout reached with fullscreen window. Postponing...";
+            return;
```
Comment (palinek) [inline src/idlenesswatcher.cpp:115]: How about a situation when the fullscreen application ends w/o user interaction? Then we will never trigger the idle action. Shouldn't we at least reset the idleness here (`KIdleTime::simulateUserActivity`) to get another `timeoutReached` after a while?

Comment (adjamhub) [reply src/idlenesswatcher.cpp:115]: Your suggestion is probably the safest path. I'm going to update my patch following it. Thanks :)

**Review by palinek — APPROVED** [review]
GTM

File: src/idlenesswatcher.cpp [hunk 2]
```diff
@@ -102,8 +104,19 @@ void IdlenessWatcher::setup()
...
 
-void IdlenessWatcher::timeoutReached(int identifier)
+void IdlenessWatcher::timeoutReached(int identifier,int /*msec*/)
 {
+    // check if disable Idleness when fullscreen is enabled
+    if (mPSettings.isDisableIdlenessWhenFullscreenEnabled()) {
+        WId w = KWindowSystem::activeWindow();
+        KWindowInfo info(w, NET::WMState);
+        if (info.hasState(NET::FullScreen)) {
+            qDebug() << "idleness timeout reached with fullscreen window. Postponing...";
```
Comment (tsujan) [inline src/idlenesswatcher.cpp:114]: Please remove or comment out the qDebug() line above!

File: config/idlenesswatchersettings.cpp [hunk 3]
```diff
@@ -72,6 +72,8 @@ void IdlenessWatcherSettings::mConnectSignals()
     connect(mUi->backlightSlider, SIGNAL(valueChanged(int)), SLOT(saveSettings()));
     connect(mUi->idleTimeBacklightTimeEdit, SIGNAL(timeChanged(const QTime &)), SLOT(saveSettings()));
     connect(mUi->onBatteryDischargingCheckBox, SIGNAL(toggled(bool)), SLOT(saveSettings()));
+
+    connect(mUi->disableIdlenessFullscreenBox, SIGNAL(toggled(bool)), this, SLOT(saveSettings()));
```
Comment (tsujan) [inline config/idlenesswatchersettings.cpp:76]: Please don't use the old connection format (above and below)! The other lines should be changed later (in another PR).

File: src/idlenesswatcher.cpp (hunk 2 above)
Comment (adjamhub) [reply src/idlenesswatcher.cpp:114]: Ok!

**Comment by adjamhub** [comment]
If you are ok about, I can provide another PR about lxqt-powermanagement moving from the old signal/slot syntax to the new one.

**Comment by tsujan** [comment]
Thanks for the idea and its implementation!

> If you are ok about, I can provide another PR about lxqt-powermanagement moving from the old signal/slot syntax to the new one.

That would be nice of you!

**Pull request merged**